test/compile_test/interpolators_makima_incl_test.cpp

```diff
@@ -0,0 +1,9 @@
+//  Copyright Matt Borland 2021.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// A sanity check that this file
+// #includes all the files that it needs to.
+//
+#include <boost/math/interpolators/makima.hpp>
```

Comment:
Instantiation?

Reply:
Something like:

```
// we can't help but include a std lib header here:
#include <vector>

int main()
{
  std::vector<double> x{1, 5, 9 , 12};
  std::vector<double> y{8,17, 4, -3};
  using boost::math::interpolators::makima;
  makima<std::vector<double>> spline(std::move(x), std::move(y));
  // evaluate at a point:
  double z = spline(3.4);
  // evaluate derivative at a point:
  double zprime = spline.prime(3.4);
}
```

Which is straight out of the docs, might do it - I hope!!

The other interpolators all have the same interface... so hopefully just change the name ;)

Reply:
I should have mentioned I skipped the interpolators. I am sure they could be made to work with a c-array or pointers, but not sure it was worth the effort. Adding `std::vector` should work just fine.